digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/vvh/VVHClient.scala

```diff
@@ -522,11 +522,13 @@ class VVHRoadLinkClient(vvhRestApiEndPoint: String) extends VVHClientOperations
     * In Use - 0
     * Under Construction - 1
     * Planned - 3
+    * Temporarily Not In Use - 4
+    * Expiring Soon - 5
     */
   protected def roadLinkStatusFilter(feature: Map[String, Any]): Boolean = {
     val attributes = feature("attributes").asInstanceOf[Map[String, Any]]
-    val linkStatus = extractAttributes(attributes).getOrElse("CONSTRUCTIONTYPE", BigInt(0)).asInstanceOf[BigInt]
-    linkStatus == ConstructionType.InUse.value || linkStatus == ConstructionType.Planned.value || linkStatus == ConstructionType.UnderConstruction.value
+    val linkStatus: ConstructionType = extractAttributes(attributes).getOrElse("CONSTRUCTIONTYPE", ConstructionType.InUse).asInstanceOf[ConstructionType]
+    ConstructionType.filteredLinkStatus.contains(linkStatus)
```

Comment:
I likes this version a little bit more clean, instead of chaining all of the different TYPES with OR operation.
Although, I'm not sure it will work - if any of you knows best about how Scala handles objects and stuff and see a problem there, let me know.

Reply:
I think this is good.